crates/oxc_query/src/vertex.rs

```diff
@@ -279,15 +279,21 @@ impl<'a> Vertex<'a> {
     }
 
     pub fn function_scope_flag(&self, adapter: &Adapter<'_>) -> ScopeFlags {
-        let wanted_node_hash = match &self {
-            Vertex::ArrowFunction(data) => calculate_hash(data.arrow_expression),
-            Vertex::FnDeclaration(data) => calculate_hash(data.function),
+        let (wanted_node_hash, wanted_node_span) = match &self {
+            Vertex::ArrowFunction(data) => (calculate_hash(data.arrow_expression), data.arrow_expression.span),
+            Vertex::FnDeclaration(data) => (calculate_hash(data.function), data.function.span),
             _ => unreachable!(
                 "'function_scope_flag' function should only ever be called with an ArrowFunction or FnDeclaration"
             ),
         };
 
         let found = adapter.semantic.nodes().iter().find(|x| {
+            let span = x.kind().span();
+
+            if span.start != wanted_node_span.start || span.end != wanted_node_span.end {
+                return false;
+            }
+
             let hash_of_node = match x.kind() {
                 AstKind::ArrowExpression(ae) => calculate_hash(ae),
                 AstKind::Function(fn_) => calculate_hash(fn_),
```

Comment:
<i class='graphite__hidden'>[Re: lines 302 to 302]<br/></i>

Do we even need to hash here? Can two functions have the same span?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1527?utm_source=unchanged-line-comment">Graphite</a>.</span>